when encountering error during fetch return "" in web_base.py

oegedijk opened the pull request:
when e.g. downloading a sitemap with a malformed url (e.g.  "ttp://example.com/index.html" with the h omitted at the beginning of the url), this will ensure that the sitemap download does not crash, but just emits a warning. (maybe should be optional with e.g. a `skip_faulty_urls:bool=True` parameter, but this was the most straightforward fix)

@rlancemartin, @eyurtsev

<!-- Thank you for contributing to LangChain!

Replace this comment with:
  - Description: a description of the change, 
  - Issue: the issue # it fixes (if applicable),
  - Dependencies: any dependencies required for this change,
  - Tag maintainer: for a quicker response, tag the relevant maintainer (see below),
  - Twitter handle: we announce bigger features on Twitter. If your PR gets announced and you'd like a mention, we'll gladly shout you out!

Please make sure you're PR is passing linting and testing before submitting. Run `make format`, `make lint` and `make test` to check this locally.

If you're adding a new integration, please include:
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use.

Maintainer responsibilities:
  - General / Misc / if you don't know who to tag: @baskaryan
  - DataLoaders / VectorStores / Retrievers: @rlancemartin, @eyurtsev
  - Models / Prompts: @hwchase17, @baskaryan
  - Memory: @hwchase17
  - Agents / Tools / Toolkits: @hinthornw
  - Tracing / Callbacks: @agola11
  - Async: @agola11

If no one reviews your PR within a few days, feel free to @-mention the same people again.

See contribution guidelines for more information on how to write/run tests, lint, etc: https://github.com/hwchase17/langchain/blob/master/.github/CONTRIBUTING.md
 -->

oegedijk committed:
when encountering error during fetch return "" in web_base.py

when e.g. downloading a sitemap with a malformed url (e.g.  "ttp://example.com/index.html" with the h omitted at the beginning of the url), this will ensure that the sitemap download does not crash, but just emits a warning.

baskaryan commented:
yea i think would be nice to make this optional with a `silent_errors` or `continue_on_failure` flag

oegedijk committed:
adds continue_on_failure flag to WebBaseLoader

oegedijk committed:
adds continue_on_failure=True suggestion

when encountering a failure loading a url from a list of urls,
will suggest setting the continue_on_failure flag

oegedijk committed:
adds continue_on_failure to SitemapLoader

oegedijk committed:
clean up warning and exception message

oegedijk committed:
adds continue_on_failure to blackboard and gitbook

oegedijk commented:
Went for the `continue_on_failure` flag and defaulted it to False to maintain current behaviour. 

When encountering an error now, added a suggestion to the logger to set the flag to True. 

Added it Sitemap, Gitbook and Blackboard loaders as they all derive from `WebBaseLoader`, while overwriting the `__init__`.

hwchase17 reviewed (approved):
nice thanks!

oegedijk committed:
black formatting of changes

oegedijk commented:
added black formatting of changed files

baskaryan committed:
fmt
